crates/rust-analyzer/src/global_state.rs

```diff
@@ -87,7 +87,7 @@ pub(crate) struct GlobalState {
     pub(crate) flycheck_sender: Sender<flycheck::Message>,
     pub(crate) flycheck_receiver: Receiver<flycheck::Message>,
     pub(crate) last_flycheck_error: Option<String>,
-    pub(crate) diagnostics_received: bool,
+    pub(crate) flycheck_status: FxHashMap<usize, FlycheckStatus>,
```

Comment:
```suggestion
    pub(crate) flycheck_status: Vec<FlycheckStatus>,
```
might as well be just a vec that we pre allocate to unknown when we setup the flycheck workers, though I think I agree with you that encoding the state into the flycheck worker might be better? Then it can send out a corresponding clear event when necessary.

Reply:
I tried changing it to a vector, but I didn't like all the unwraps and rechecking that the vec really is large enough else extend it.

So I tried putting it inside the flycheck actor.